### PR TITLE
nixos/security.pam: remove `with lib;`

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #208242

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by philiptaron — CHANGES_REQUESTED** [review]
Test this with the following command:

```
nix-build --expr 'with import ./. { }; lib.filterAttrs (name: value: lib.hasPrefix "pam" name) nixosTests'
```

File: nixos/modules/security/pam.nix [hunk 1]
```diff
@@ -1168,9 +1164,9 @@ in
         '';
       };
 
-      control = mkOption {
+      control = lib.mkOption {
         default = "sufficient";
-        type = types.enum [ "required" "requisite" "sufficient" "optional" ];
+        type = lib.types.enum [ "required" "requisite" "sufficient" "lib.optional" ];
```
Comment (philiptaron) [inline nixos/modules/security/pam.nix:1169]: this is not right

File: nixos/modules/security/pam.nix [hunk 2]
```diff
@@ -1345,18 +1341,18 @@ in
...
           If set, then the authenticating user must be a member of this group
           to use this module.
         '';
       };
 
-      control = mkOption {
+      control = lib.mkOption {
         default = "sufficient";
-        type = types.enum [ "required" "requisite" "sufficient" "optional" ];
+        type = lib.types.enum [ "required" "requisite" "sufficient" "lib.optional" ];
```
Comment (philiptaron) [inline nixos/modules/security/pam.nix:1355]: This is not right

File: nixos/modules/security/pam.nix [hunk 3]
```diff
@@ -1386,9 +1382,9 @@ in
           More information can be found [here](https://developers.yubico.com/yubico-pam/).
         '';
       };
-      control = mkOption {
+      control = lib.mkOption {
         default = "sufficient";
-        type = types.enum [ "required" "requisite" "sufficient" "optional" ];
+        type = lib.types.enum [ "required" "requisite" "sufficient" "lib.optional" ];
```
Comment (philiptaron) [inline nixos/modules/security/pam.nix:1387]: this is not right

File: nixos/modules/security/pam.nix [hunk 4]
```diff
@@ -1580,30 +1576,30 @@ in
...
-        attrValues
+        lib.attrValues
         (catAttrs "rules")
-        (concatMap attrValues)
-        (concatMap attrValues)
-        (filter (rule: rule.enable))
+        (lib.concatMap lib.attrValues)
+        (lib.concatMap lib.attrValues)
+        (lib.filter (rule: rule.enable))
         (catAttrs "modulePath")
```
Comment (philiptaron) [inline nixos/modules/security/pam.nix:1594]: wonder why this didn't need `lib.`...

**Review by philiptaron — CHANGES_REQUESTED** [review]
Couple more defects.

File: nixos/modules/security/pam.nix (hunk 4 above)
Comment (Stunkymonkey) [reply nixos/modules/security/pam.nix]: I added it as well.

**Review by philiptaron — APPROVED** [review]
`nix-build -A nixosTests.pam-file-contents` fails on master, too.

**Pull request merged**